daskhub/values.yaml

```diff
@@ -52,6 +39,7 @@ jupyterhub:
       tag: 2020.07.28
     defaultUrl: "/lab"  # Use jupyterlab by defualt.
     extraEnv:
+      DASK_GATEWAY__ADDRESS: "http://proxy-public/services/dask-gateway"  # Internal address to connect to the Dask Gateway.
       DASK_GATEWAY__PUBLIC_ADDRESS: "/services/dask-gateway/"  # Sets the Dask dashboard link in cluster and client reprs.
       DASK_GATEWAY__AUTH__TYPE: "jupyterhub"  # Use JupyterHub to authenticate with Dask Gateway.
```

Comment:
I note that you are setting c.KubeSpawner.environment in hub.extraConfig, but that you are also setting some in singleuser.extraEnv. I suggest that you set everything in hub.extraEnv through c.KubeSpawner.environment to not cause confusion by spreading this configuration to multiple locations that may conflict if not updated in sync.

__ADDRESS is set in both c.KubeSpawner.environment and singleuser.extraEnv while it would suffice with one.
__PROXY_ADDRESS is set only in c.KubeSpawner.environment
__PUBLIC_ADDRESS is only set in singleuser.extraEnv
__PUBLIC_ADDRESS is only set in singleuser.extraEnv
__AUTH_TYPE is only set in singleuser.extraEnv

I conclude that you need to set __PROXY_ADDRESS in the hub.extraConfig code because you wanted to reference release_name and release_namespace to construct the string.

---

Btw, if you want, you could reference other environment variables if they are set. But for this strategy we should have 
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1747 resolved first.

```yaml
jupyterhub:
  singleuser:
    extraEnv:
      # This method assumes there is environment variables named $RELEASE_NAME
      # and $RELEASE_NAMESPACE explicitly set somewhere else!
      DASK_GATEWAY__PROXY_ADDRESS: gateway://traefik-$(RELEASE_NAME)-dask-gateway.$(RELEASE_NAMESPACE):80
```

Reply:
> I conclude that you need to set __PROXY_ADDRESS in the hub.extraConfig code because you wanted to reference release_name and release_namespace to construct the string.

Yep, that's right. I put the "static" stuff in `jupyterhub.singleuser.extraEnv` and the dynamic variables in the `hub.extraConfig`. Having it all in one spot is probably best.